enhancements/cloud-integration/azure/azure-workload-identity.md

```diff
@@ -0,0 +1,340 @@
+- Creating Managed Identity infrastructure with federated credentials for OpenShift operator `ServiceAccounts` (identified by namespace & name) and to output secrets containing the `clientID` of the Managed Identity to be provided as manifests for the installer. This command will process `CredentialsRequest` custom resources to identify service accounts that will be associated with Managed
+  Identities in Azure as federated credentials. For self-managed installation, `CredentialsRequests` will be exracted from the release image.
+
+```sh
+$ ./ccoctl azure
+Creating/updating/deleting cloud credentials objects for Azure
+
+Usage:
+  ccoctl azure [command]
+
+Available Commands:
+  create-all                Create key pair, identity provider and Azure Managed Identities
+  create-identity-provider  Create identity provider
+  create-key-pair           Create a key pair
+  create-managed-identities Create Azure Managed Identities
+  delete                    Delete Azure identity provider and Managed Identity infrastructure
+
+Flags:
+  -h, --help   help for azure
+
+Use "ccoctl azure [command] --help" for more information about a command.
+```
+
+#### Credentials secret
+
+OpenShift operators currently obtain their long-lived credentials from a config secret with the following format:
+
+```yaml
+apiVersion: v1
+data:
+  azure_client_id: <client id>
+  azure_client_secret: <client secret>
+  azure_region: <region>
+  azure_resource_prefix: <resource group prefix eg. "abutcher-az-t68n4">
+  azure_resourcegroup: <resource group eg. "abutcher-az-t68n4-rg">
+  azure_subscription_id: <subscription id>
+  azure_tenant_id: <tenant id>
+kind: Secret
+type: Opaque
+```
+
+We propose that when utilizing Azure Workload Identity, the credentials secret will contain an `azure_client_id` that is the `clientID` of the Managed Identity provisioned by `ccotcl` for the operator. The `azure_client_secret` key will be absent and instead we can provide the path to the mounted `ServiceAccount` token as an `azure_federated_token_file` key; the path to the mounted token is well
+known and is specified in the operator deployment.
+
+The resource group in which the installer will create infrastructure will not be known when these secrets are generated by `ccoctl` ahead of installation and operators which rely on `azure_resourcegroup` and `azure_resource_prefix` such as the
+[image-registry](https://github.com/openshift/cluster-image-registry-operator/blob/8556fd48027f89e19daad36e280b60eb93d012d4/pkg/storage/azure/azure.go#L95-L100) should obtain the resource group details from the cluster `Infrastructure` object instead.
```

Comment:
How are OpenShift operators with Azure SDKs consuming the credentials today? I assume the operator is configuring the SDK client based on the contents of the secret vs the SDK reading the secret as a configuration? I assume its the former?